### PR TITLE
feature(Products Search Endpoint): Create product search by name endp…

### PR DESCRIPTION
**Pull request by patel-prakashkumar**
…oint

# Description

Please provide a short summary explaining what this PR is about.

Fixes # (issue)

# How to test?

Please provide a short summary how your changes can be tested?

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [ ] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [ ] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by Yuliia-Balandiuk — DISMISSED** [review]

File: packages/client/src/components/ProductListDropDown/ProductListDropDown.component.js [hunk 1]
```diff
@@ -15,10 +15,8 @@ const ProductListDropdown = ({ options, ...props }) => {
     </div>
   );
 };
-
 ProductListDropdown.propTypes = {
-  options: PropTypes.string.isRequired,
-  label: PropTypes.string.isRequired,
+  options: PropTypes.arrayOf(PropTypes.string).isRequired,
```
Comment (shpomp) [inline packages/client/src/components/ProductListDropDown/ProductListDropDown.component.js:19]: This change does not belong in this PR.

File: packages/server/api/routes/products.router.js [hunk 2]
```diff
@@ -41,6 +41,37 @@ router.get('/', (req, res, next) => {
...
   }
 });
+/**
+ * @swagger
+ * /products/search:
+ *  get:
+ *    tags:
+ *    - products
+ *    summary: Get all product Name Search
+ *    description:
```
Comment (shpomp) [inline packages/server/api/routes/products.router.js:51]: As the Sprint2 is stricter, I will ask for some naming changes as well 🙏🏽 

 *    summary: Search products by name
 *    Will return all products where the name matches the search input.
 *  etc.

File: packages/server/api/routes/products.router.js [hunk 3]
```diff
@@ -57,9 +88,8 @@ router.get('/', (req, res, next) => {
  *       name: ID
  *       schema:
  *         type: integer
- *         required: true
+ *         required: flase
```
Comment (shpomp) [inline packages/server/api/routes/products.router.js:91]: typo

File: packages/server/api/controllers/products.controller.js [hunk 4]
```diff
@@ -52,7 +53,26 @@ const getProductsByCategory = async (category) => {
   }
 };
 
+// Get Search product by Name
+const getProductBySearch = async (searchText, res) => {
+  if (!searchText) {
+    res.json([]);
+    return;
+  }
```
Comment (shpomp) [inline packages/server/api/controllers/products.controller.js:61]: This would more often be included under the '/' route and treat the search input as a query. 
If you want to keep it separate, it should not return an empty array for no search input. If it does, then you cannot just use this endpoint for the Collections page and you will have to combine it with the GET all products anyway - that is not very practical. 
In general, I don't see any good reason for this to be a separate endpoint, but maybe you do 😉

**Review by shpomp — CHANGES_REQUESTED** [review]

File: packages/server/api/routes/products.router.js (hunk 2 above)
Comment (patel-prakashkumar) [reply packages/server/api/routes/products.router.js:51]: thank you for feedback. I will update.

File: packages/server/api/controllers/products.controller.js (hunk 4 above)
Comment (patel-prakashkumar) [reply packages/server/api/controllers/products.controller.js:61]: I have created separate endpoint bcz I having difficulty to making swagger Api documentation now I find solution so I will add search query with category query.

File: packages/client/src/components/ProductListDropDown/ProductListDropDown.component.js (hunk 1 above)
Comment (patel-prakashkumar) [reply packages/client/src/components/ProductListDropDown/ProductListDropDown.component.js:19]: I will remove. Its also from my previous Pr.

Comment (shpomp) [reply packages/client/src/components/ProductListDropDown/ProductListDropDown.component.js]: Why is this resolv ed if it is not removed? @patel-prakashkumar

Comment (patel-prakashkumar) [reply packages/client/src/components/ProductListDropDown/ProductListDropDown.component.js]: Resolved issue

**Review by shpomp — APPROVED** [review]

**Review by 27ManishaPatel — APPROVED** [review]
